scr/src/SCRSubAgent.cc

```diff
@@ -112,5 +113,17 @@ SCRSubAgent::mount (SCRAgent *parent)
 void
 SCRSubAgent::unmount ()
 {
-	my_comp = 0;
+    // In general, a Y2Component is owned by its Y2ComponentCreator.
+    // In practice, only some subclasses of Y2ComponentCreator
+    // delete the components they own; the others leak.
+    // Also it prevents UnmountAgent from working.
+    // Here we cheat and steal Y2ProgramComponent from Y2CCProgram,
+    // which is fine since that class does not delete the component it creates.
```

Comment:
Um, should we add a comment regarding this into the `Y2CCProgram` as well?

Reply:
Ah, good point. Added.